egov-pg-service/src/main/java/org/egov/pg/service/PaymentsService.java

```diff
@@ -0,0 +1,118 @@
+package org.egov.pg.service;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.egov.pg.config.AppProperties;
+import org.egov.pg.models.CollectionPayment;
+import org.egov.pg.models.CollectionPaymentDetail;
+import org.egov.pg.models.CollectionPaymentRequest;
+import org.egov.pg.models.CollectionPaymentResponse;
+import org.egov.pg.models.TaxAndPayment;
+import org.egov.pg.models.enums.CollectionPaymentModeEnum;
+import org.egov.pg.repository.ServiceCallRepository;
+import org.egov.pg.web.models.TransactionRequest;
+import org.egov.tracer.model.CustomException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+public class PaymentsService {
+	
+	@Autowired
+	private ServiceCallRepository repository;
+	
+	@Autowired
+	private AppProperties props;
+	
+	public CollectionPayment registerPayment(TransactionRequest request) {
+		CollectionPayment payment = getPaymentFromTransaction(request);
+		payment.setInstrumentDate(request.getTransaction().getAuditDetails().getCreatedTime());
+		payment.setInstrumentNumber(request.getTransaction().getTxnId());
+		payment.setTransactionNumber(request.getTransaction().getTxnId());
+		
+		CollectionPaymentRequest paymentRequest = CollectionPaymentRequest.builder()
+				.requestInfo(request.getRequestInfo()).payment(payment).build();
+		StringBuilder builder = new StringBuilder();
```

Comment:
Use string concat instead

Reply:
Isn't it a bad practice to use String concat? It creates too many unnecessary strings in the pool, StringBuilder just creates one final string in the pool saving memory

Reply:
StringBuilder is only useful in loops, everywhere else best avoided, in your case there's barely 4-5 fixed concats. 
https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java